### PR TITLE
Prespark2

### PR DESCRIPTION
**Pull request by cseed**
I'd like to move master to Spark 2 and Scala 2.11.

These changes get us as close as possible.  They include:

 - remove SparkExport, use reflection to get path of partition when loading from parquet
 - remove SparkManager
 - upgrade to Kudu 1.1.0 (Spark 2 support)

The distance between this and Spark 2 is very small, see https://github.com/hail-is/hail/commit/95a588cfa72391d4303bf6891fd017ec211989db.

When the master moves to Spark 2, we can maintain a spark1 branch until the on-prem machines get upgraded.  Ideally, the spark1 branch could get rebased automatically as part of the CI, although I'm not quite sure how we'd handle conflicts.  Alternatively, we could maintain a spark2 -> spark1 diff in the repo that gets applied as part of testing.

Fixes https://github.com/hail-is/hail/issues/1117

### CODE REVIEW

File: src/main/scala/org/broadinstitute/hail/utils/richUtils/RichSQLContext.scala [hunk 1]
```diff
@@ -22,26 +19,7 @@ class RichSQLContext(val sqlContext: SQLContext) extends AnyVal {
     val rdd = df.rdd
 
     val oldIndices = rdd.partitions
-      .map { p =>
-        val parquetInputSplit = SparkExport.sqlNewHadoopPartitionRawSplit(p)
-
-        /*
-         * Use reflection to invoke getPath instead of importing ParquetInputSplit and casting since the parquet
-         * package (parquet vs org.apache.parquet) moved and depends on the distribution we are running against.
-         */
```
Comment (danking) [inline src/main/scala/org/broadinstitute/hail/utils/richUtils/RichSQLContext.scala]: I would copy this back up to the new code as well.

File: build.gradle [hunk 2]
```diff
@@ -76,8 +89,8 @@ dependencies {
     compile 'org.scalanlp:breeze-natives_' + scalaMajorVersion + ':0.11.2'
```
Comment (danking) [inline build.gradle]: Let's add another variable for `breezeVersion` and set it to `0.12` for the 2.0 case.

**Comment by tomwhite** [comment]
@cseed, not sure if you know, but you can actually run both Spark 1 and 2 on CDH side-by-side (I assume that's what you mean by the on-prem machines). See here for how to install: https://www.cloudera.com/documentation/betas/spark2/latest/topics/spark2_installing.html

**Comment by cseed** [comment]
Thanks @tomwhite, that's great!  Unfortunately, we have a second local cluster with another distribution on Spark 1.5 that won't get upgrade until early 2017 at the earliest.

**Comment by tomwhite** [comment]
@cseed, ah sorry to hear that!

**Comment by cseed** [comment]
@danking Can you take another look?

File: build.gradle [hunk 3]
```diff
@@ -48,9 +57,15 @@ if (spark2) {
...
     kuduSpark = 'spark'
-    py4jVersion = '0.9'
+    if (sparkVersion ==~ /^1\.5.*/) {
+        py4jVersion = '0.8.2.1'
+    } else {
+        py4jVersion = '0.9' // 1.6
+    }
 }
 
+println(py4jVersion)
```
Comment (danking) [inline build.gradle]: Do we need this?

**Pull request merged**